staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -682,8 +682,10 @@ type GroupVersionForDiscovery struct {
 
 // APIResource specifies the name of a resource and whether it is namespaced.
 type APIResource struct {
-	// name is the name of the resource.
+	// name is the plural name of the resource.
 	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+	// singular is the singular name of the resource.  This allows clients to handle plural and singular opaquely.
```

Comment:
Add to documentation where the plural name is used vs where the singular name is used.